component/common/appendable/appendable.go

```diff
@@ -79,11 +79,7 @@ func (app *flowAppender) Append(ref storage.SeriesRef, l labels.Labels, t int64,
 	} else {
 		metrics.GlobalRefMapping.RemoveStaleMarker(uint64(ref))
 	}
-	app.buffer[t] = append(app.buffer[t], &metrics.FlowMetric{
-		GlobalRefID: uint64(ref),
-		Labels:      l,
-		Value:       v,
-	})
+	app.buffer[t] = append(app.buffer[t], metrics.NewFlowMetric(uint64(ref), l, v))
```

Comment:
Thinking aloud: 

One of the nice things about the storage.Appendable API is that it runs on the stack: you don't have to allocate a new pointer for each sample when you send it to the WAL. 

Since the buffer is a slice and will exist in the heap anyway, there might be some nice improvements in allocations by making it store a non-pointer value so there's only one big thing to clean up instead of many small pointers.

Our profiles aren't showing that this is a problem right now, so it's not something I'd spend time on immediately, and it's also probably not worth working on while the API is still being shaped out.